packages/terra-table/src/Table.jsx

```diff
@@ -1,68 +1,425 @@
-import TableRow from './TableRow';
-import TableCell from './TableCell';
-import SelectableTableRows from './SelectableTableRows';
-import TableSingleSelectableRows from './SingleSelectableRows';
-import TableMultiSelectableRows from './MultiSelectableRows';
-import TableSubheader from './TableSubheader';
+import ContentContainer from 'terra-content-container';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
 import styles from './Table.module.scss';
+import sectionShape from './proptypes/sectionShape';
+import headerShape from './proptypes/headerShape';
+import widthShape from './proptypes/widthShape';
+
+import Row from './subcomponents/_Row';
+import Cell from './subcomponents/_Cell';
+import Section from './subcomponents/_Section';
+import HeaderRow from './subcomponents/_HeaderRow';
+import HeaderCell from './subcomponents/_HeaderCell';
+import ChevronCell from './subcomponents/_ChevronCell';
+import CheckMarkCell from './subcomponents/_CheckMarkCell';
+import HeaderChevronCell from './subcomponents/_HeaderChevronCell';
+import HeaderCheckMarkCell from './subcomponents/_HeaderCheckMarkCell';
 
 const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * The children passed to the component
+   * The check mark styling to apply.
+   */
+  checkStyle: PropTypes.oneOf([
+    'none',
+    'icon',
+    'toggle',
+  ]),
+  /**
+   * The divider styling to apply to the child rows.
+   */
+  dividerStyle: PropTypes.oneOf(['none', 'vertical', 'horizontal', 'both']),
+  /**
+   * The width value structures associated to each column.
+   */
+  columnWidths: PropTypes.arrayOf(widthShape),
+  /**
+   * Whether or not the rows should have chevrons applied.
+   */
+  hasChevrons: PropTypes.bool,
+  /**
+   * The data to build header cells and columns.
+   */
+  headerData: headerShape,
+  /**
+   * Element to append to the top of the table. i.e. toolbars etc.
+   */
+  headerNode: PropTypes.node,
+  /**
+   * Whether or not the table should expanded to fill its parent element.
+   */
+  fill: PropTypes.bool,
+  /**
+   * Element to append to the bottom of the table. i.e. toolbars etc.
    */
-  children: PropTypes.node.isRequired,
+  footerNode: PropTypes.node,
   /**
-   * Whether or not the rows should be zebra striped
+   * The numberOfColumns to be used as a descriptor for assistive technology.
    */
-  isStriped: PropTypes.bool,
+  numberOfColumns: PropTypes.number.isRequired,
   /**
-   * Whether or not the table cells should be padded
+   * This value is used for accessibility when paged/virtualized rows are used.
+   * By default this value is derived from the number of rows passed within the sectionData.
    */
-  isPadded: PropTypes.bool,
+  numberOfRows: PropTypes.number,
+  /**
+   * The padding styling to apply to the cell content.
+   */
+  paddingStyle: PropTypes.oneOf(['none', 'standard', 'compact']),
+  /**
+   * The interaction styling to apply to the row.
+   */
+  rowStyle: PropTypes.oneOf([
```

Comment:
Might be good to describe the difference between toggle and disclose.

Reply:
Updated.